db/app/models/article-models.js

```diff
@@ -0,0 +1,42 @@
+const db = require("../../connection");
+
+function selectArticleById(id) {
+  const queryStr = `SELECT * FROM articles WHERE article_id = $1`;
+  return db.query(queryStr, [id]).then((response) => {
+    if (response.rowCount === 0) {
+      return Promise.reject({ status: 404, msg: "Not Found" });
+    }
+    return response.rows[0];
+  });
+}
+
+function selectAllArticles(sort_by='created_at', order='desc'){
+
+    const acceptedSortBy = ['created_at']
+    const acceptedOrder = ['desc', 'asc']
+
+    if (!acceptedSortBy.includes(sort_by) || !acceptedOrder.includes(order)){
+        return Promise.reject({status:400, msg: 'Bad Request'})
+    }
+
+
+    const queryStr = `SELECT articles.author, title, topic, articles.article_id, articles.created_at, articles.votes, article_img_url, COUNT(comments.article_id) AS comment_count 
```

Comment:
Just a heads up that your comment_count value will be a string. This is fine but it's just something you will need to remember when dealing with it, once you get to frontend.